src/RemoteInboxNotifications/DataSourcePoller.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * Handles polling and storage of specs
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Specs data source poller class.
+ * This handles polling specs from JSON endpoints, and
+ * stores the specs in to the database as an option.
+ */
+class DataSourcePoller {
+	const DATA_SOURCES = array(
+		'http://two.wordpress.test/rinds-specs.json',
+	);
+
+	/**
+	 * Reads the data sources for specs and persists those specs.
+	 */
+	public static function read_specs_from_data_sources() {
+		$specs = array();
+
+		// Note that this merges the specs from the data sources based on the
+		// slug - last one wins.
+		foreach ( self::DATA_SOURCES as $url ) {
+			$specs_from_data_source = self::read_data_source( $url, $specs );
+			self::merge_specs( $specs_from_data_source, $specs );
+		}
+
+		// Persist the specs as an option.
+		update_option( RemoteInboxNotificationsEngine::SPECS_OPTION_NAME, $specs );
+	}
+
+	/**
+	 * Read a single data source and return the read specs
+	 *
+	 * @param string $url The URL to read the specs from.
+	 *
+	 * @return array The specs that have been read from the data source.
+	 */
+	private static function read_data_source( $url ) {
+		$response = wp_remote_get( $url );
+
+		if ( is_wp_error( $response ) ) {
+			return [];
+		}
+
+		$body  = $response['body'];
```

Comment:
Wow that's interesting. I've added a check for the body, thanks.

Reply:
In case you're unaware, WP has some helper methods for this like [`wp_remote_retrieve_body`](https://developer.wordpress.org/reference/functions/wp_remote_retrieve_body/).